core/adapters/http_test.go

```diff
@@ -304,3 +304,225 @@ func TestQueryParameters(t *testing.T) {
 		})
 	}
 }
+
+func TestExtendedPath(t *testing.T) {
+	t.Parallel()
+
+	baseUrl := "http://example.com"
+
+	cases := []struct {
+		name         string
+		startingUrl  string
+		path         string
+		wantErrored  bool
+		expectedPath adapters.ExtendedPath
+		expectedURL  string
+	}{
+		{
+			"empty",
+			baseUrl,
+			`""`,
+			false,
+			adapters.ExtendedPath{""},
+			baseUrl,
+		},
+		{
+			"two paths",
+			baseUrl,
+			`"one/two"`,
+			false,
+			adapters.ExtendedPath{
+				"one",
+				"two",
+			},
+			"http://example.com/one/two",
+		},
+		{
+			"existing path no trailing slash",
+			"http://example.com/one",
+			`"two/three"`,
+			false,
+			adapters.ExtendedPath{
+				"two",
+				"three",
+			},
+			"http://example.com/one/two/three",
+		},
+		{
+			"existing path with trailing slash",
+			"http://example.com/one/",
+			`"two/three"`,
+			false,
+			adapters.ExtendedPath{
+				"two",
+				"three",
+			},
+			"http://example.com/one/two/three",
+		},
+		{
+			"input as arrays",
+			baseUrl,
+			`["one","two"]`,
+			false,
+			adapters.ExtendedPath{
+				"one",
+				"two",
+			},
+			"http://example.com/one/two",
+		},
+		{
+			"input begins with slash",
+			baseUrl,
+			`"/one/two"`,
+			false,
+			adapters.ExtendedPath{
+				"",
+				"one",
+				"two",
+			},
+			"http://example.com/one/two",
+		},
+		{
+			"input ends with slash",
+			baseUrl,
+			`"one/two/"`,
+			false,
+			adapters.ExtendedPath{
+				"one",
+				"two",
+				"",
+			},
+			"http://example.com/one/two",
+		},
+		{
+			"bad array input",
+			baseUrl,
+			`["one","two"`,
+			true,
+			adapters.ExtendedPath{},
+			baseUrl,
+		},
+	}
+
+	for _, tt := range cases {
+		test := tt
+		t.Run(test.name, func(t *testing.T) {
+			ep := adapters.ExtendedPath{}
+			err := json.Unmarshal([]byte(test.path), &ep)
+			hga := adapters.HTTPGet{
+				URL:     cltest.WebURL(t, test.startingUrl),
+				ExtPath: ep,
+			}
+			hpa := adapters.HTTPPost{
+				URL:     cltest.WebURL(t, test.startingUrl),
+				ExtPath: ep,
+			}
+			if test.wantErrored {
```

Comment:
To continue our effort of reducing `if`'s in table driven tests, let's take the single `wantErrored` `true` test and place that in a separate test, with or without its own table. This will set a better precedent for the growth of these tests.

Resulting two test name suggestions:
1. TestExtendedPath_Success
2. TestExtendedPath_Error